File: C++/Problems/Problems/problem_1310.cpp
#include "problem_1310.h"

bool problem_1310::test() {
    std::vector arr { 1, 3, 4, 8 };
    std::vector<std::vector<int>> queries  {  { 0, 1 },  { 1, 2 },  { 0, 3 },  { 3, 3 } };

    std::vector expected =  { 2, 7, 14, 8 };

    auto result = xorQueries(arr, queries);

    print1DVector(result);

    return expected == result;
}

std::vector<int> problem_1310::xorQueries(std::vector<int> &arr, std::vector<std::vector<int>> &queries) {
    int n = arr.size();
    std::vector<int> pref(n + 1);
    pref[0] = 0;
    for (int i = 0; i < n; ++i)
        pref[i + 1] = pref[i] ^ arr[i];
    int m = queries.size();
    std::vector<int> result(m);
    for (int i = 0; i < m; ++i)
    {
        result[i] = pref[queries[i][0]] ^ pref[queries[i][1] + 1];
    }

    return result;
}

File: Python/Problems/Problems/problem1310.py
import heapq
import math
import queue
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter, deque

from .problem import Problem

class Problem1310(Problem):
    def test(self) -> bool:
        arr = [1, 3, 4, 8]
        queries = [[0, 1], [1, 2], [0, 3], [3, 3]]

        expected = [2, 7, 14, 8]

        result = self.xorQueries(arr, queries)

        print1DArray(result)

        return expected == result

    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        n = len(arr)
        pref = [0]
        for i in range(n):
            pref.append(pref[i] ^ arr[i])
        m = len(queries)
        result = []
        for i in range(m):
            result.append(pref[queries[i][0]] ^ pref[queries[i][1] + 1])

        return result